File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Deque<Character> que = new ArrayDeque<>();
        
        for(int i=0; i<s.length(); i++){
            if(!que.isEmpty() && que.peekLast() == s.charAt(i)){
                que.pollLast();
            }else{
                que.add(s.charAt(i));
            }
        }        
        
        if(que.isEmpty()){
            return 1;
        }else{
            return 0;
        }

    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    const que = [];
    
    for(let i=0; i<s.length; i++){
        if(que.length && que[que.length-1] == s.charAt(i)){
            que.pop();
        }else que.push(s.charAt(i));
    }
    
    if(que.length){
        return 0;
    }else return 1;

}